src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1864,13 +1864,22 @@ def aks_kollect(cmd,    # pylint: disable=too-many-statements,too-many-locals
     print()
     print("Starts collecting diag info for cluster %s " % name)
 
+    # Form containerName from fqdn, as it was previously jsut the location of code is changed.
+    # https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#container-names
+    maxContainerNameLength = 63
+    fqdn = mc.fqdn if mc.fqdn is not None else mc.private_fqdn
+    normalized_container_name = fqdn.replace('.', '-')
+    len_of_container_name = normalized_container_name.index("-hcp-")
```

Comment:
I am not sure if the part in fqdn before "hcp" would be longer than 63, if so, this may break the limit.

Reply:
https://docs.microsoft.com/en-us/azure/aks/troubleshooting#what-naming-restrictions-are-enforced-for-aks-resources-and-parameters This might help.

essentially this makes sure that in fqdn for cluster we take all character before “hpc” although the doc kind of guarantees that cluster name can’t be more then 63 character but just to be safe we take 63 in case command line or direct ARM has a loop. 

Also container name cannot be more then 63 character as well.

Reply:
I see, but without the background knowledge about the limit of the cluster name, what I am concerning about is that the code might get a container name longer than 63 if the fqdn is not valid (through this should not happen).

Reply:
I think this code will only ever trigger for privatelink clusters because as the link shared above normal cluster name cannot be more then 63 but in case of privatelink cluster when they are formed the final fqdn contains “privatelink” which is nomenclature standard,

Reply:
Okay, since we have a strong restriction for the naming, I think the current implementation is all good.